ui/src/pages/namespace/Explorer/Tree/index.tsx

```diff
@@ -61,11 +63,16 @@ const ExplorerPage: FC = () => {
       </Card>
     );
 
-  const results = data?.children?.results ?? [];
-  const showTable = !isRoot || results.length > 0;
-  const noResults = isSuccess && results.length === 0;
+  const children = (data?.type === "directory" && data?.children) || [];
+  const showTable = !isRoot || children.length > 0;
+  const noResults = isSuccess && children.length === 0;
   const wideOverlay = !!previewNode;
 
+  const existingNames =
+    data?.type === "directory" && data.children
+      ? data.children?.map((file) => getFilenameFromPath(file.path))
+      : [];
+
```

Comment:
Could we use the children variable from above? 
`const existingNames = children.map((file) => getFilenameFromPath(file.path));`